gpytorch/lazy/block_diag_lazy_tensor.py

```diff
@@ -34,12 +34,14 @@ def _add_batch_dim(self, other):
         return other
 
     @cached(name="cholesky")
-    def _cholesky(self):
-        return self.__class__(self.base_lazy_tensor._cholesky())
+    def _cholesky(self, upper=False):
+        from .triangular_lazy_tensor import TriangularLazyTensor
 
-    def _cholesky_solve(self, rhs):
+        return TriangularLazyTensor(self.__class__(self.base_lazy_tensor._cholesky(upper=upper)), upper=upper)
```

Comment:
Same issue here

Reply:
I think that's okay - in some sense that's what we want (an block-triangular matrix of triangular matrices)